lib/libc/misc/lib_envpath.c

```diff
@@ -0,0 +1,284 @@
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+
+#include <tinyara/envpath.h>
+
+#include "libc.h"
+
+#if defined(CONFIG_LIB_ENVPATH)
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct envpath_s {
+	FAR char *next;				/* Pointer to the next (unterminated) value in the PATH variable */
+	char path[1];
+};
+
+#define SIZEOF_ENVPATH_S(n) (sizeof(struct envpath_s) + (n) - 1)
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: envpath_init
+ *
+ * Description:
+ *   Initialize for the traversal of each value in the PATH variable.  The
+ *   usage is sequence is as follows:
+ *
+ *   1) Call envpath_init() to initialize for the traversal.  envpath_init()
+ *      will return an opaque handle that can then be provided to
+ *      envpath_next() and envpath_release().
+ *   2) Call envpath_next() repeatedly to examine every file that lies
+ *      in the directories of the PATH variable
+ *   3) Call envpath_release() to free resources set aside by envpath_init().
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   On success, envpath_init() return a non-NULL, opaque handle that may
+ *   subsequently be used in calls to envpath_next() and envpath_release().
+ *   On error, a NULL handle value will be returned.  The most likely cause
+ *   of an error would be that there is no value associated with the PATH
+ *   variable.
+ *
+ ****************************************************************************/
+
+ENVPATH_HANDLE envpath_init(FAR const char *name)
+{
+	FAR struct envpath_s *envpath;
+	FAR char *path;
+
+	/* Get the value of the PATH variable */
+
+	path = getenv(name);
+	if (!path) {
+		/* getenv() will return a NULL value if the PATH variable does not
```

Comment:
add appropriate log specifying the type of error before returning NULL

Reply:
Ok